docs/index.bs

```diff
@@ -3136,15 +3142,18 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
       1. Let |activeWorker| be |registration|'s <a>active worker</a>.
       1. If |activeWorker|'s [=service worker/list of router rules=] is [=list/is not empty=]:
           1. Let |source| be the result of running [=Get Router Source=] algorithm with |registration|'s <a>active worker</a> and |request|.
-          1. If |source| is {{RouterSource/"network"}}, return null.
-          1. Else if |source| is {{RouterSource/"cache"}}, then:
-              1. Let |requestResponses| be the result of running [=Query Cache=] with |request|.
-              1. If |requestResponses| is an empty [=list=], return null.
-              1. Else:
-                  1. Let |requestResponse| be the first element of |requestResponses|.
-                  1. Let |response| be |requestResponse|'s response.
-                  1. If |client| is not null, |response|'s [=response/type=] is "`opaque`", and [=cross-origin resource policy check=] with |request|'s [=request/origin=], |client|, "", and |response|'s [=filtered response/internal response=] returns <b>blocked</b>, then return null.
-                  1. Return |response|.
+          1. If |source| is {{RouterSourceEnum/"network"}}, return null.
+          1. Else if |source| is {{RouterSourceEnum/"cache"}}, or |source|["{{RouterSourceDict/cacheName}}"] [=map/exists=], then:
+              1. [=map/For each=] |cacheName| &#x2192; |cache| of the [=relevant name to cache map=]:
+                  1. If |source|["{{RouterSourceDict/cacheName}}"] [=map/exists=] and |source|["{{RouterSourceDict/cacheName}}"] does not match |cacheName|, [=continue=].
+                  1. Let |requestResponses| be the result of running [=Query Cache=] with |request|, a new {{CacheQueryOptions}},and |cache|.
```

Comment:
|cache| is going to be passes as |targetStorage| in Query Cache. Not sure if |cache| is data corresponding to the `request response list`.
https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/pull/9.html#dfn-request-response-list

Reply:
In https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/pull/9.html#ref-for-dfn-request-response-list%E2%91%A3, |cache| is a `request response list`, which is set as a value of `relevant name to cache map[cacheName]`.  From that, I assumed that this |cache| is also `request response list`.

`caches.open()` returns a promise with Cache that represents a `request response list` according to https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/pull/9.html#cache-storage-open.

It is usable with `cache.matchAll()` (https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/pull/9.html#dom-cache-matchall).
The "Query Cache" explanation says that:
> If the optional argument targetStorage is omitted, set storage to the [relevant request response list](https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/pull/9.html#dfn-relevant-request-response-list).

`relevant request response list` should be `request response list` for `this` according to https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/pull/9.html#dfn-relevant-request-response-list.

Therefore, if |cache| is passed, it should be the same meaning with open the relevant cache and passing its relevant request response list.

Reply:
I had the same question. I think you understand this better than me, but one thing that makes me unsure, is that there is only one other place in the spec where "Query Cache" has its third parameter passed: "Batch Cache Operations". All others use the default.

Are we sure that "Batch Cache Operations" is the right model, instead of, e.g., matchAll?

Reply:
I picked the steps from [CacheStorage.match()](https://w3c.github.io/ServiceWorker/#cache-storage-match) and [Cache.match()](https://w3c.github.io/ServiceWorker/#cache-match), and [Cache.matchAll()](https://w3c.github.io/ServiceWorker/#cache-matchall).  I did not picked the steps in "Batch Cache Operations".
Since this PR do not instantiate `Cache`, I need to pick a relevant |targetStorage|.

As you mention, the third parameter of "Query Cache" is rarely used, and I understand your concern.
"Query Cache" is used by three locations:
- [Cache.matchAll()](https://w3c.github.io/ServiceWorker/#ref-for-query-cache)
- [Cache.keys()](https://w3c.github.io/ServiceWorker/#ref-for-query-cache%E2%91%A0)
- and [Batch Cache Operation](https://w3c.github.io/ServiceWorker/#batch-cache-operations)

Except for the Batch Cache Operation, they are used as Cache's method.  Then, |targetStorage| is filled at https://w3c.github.io/ServiceWorker/#ref-for-dfn-relevant-request-response-list%E2%91%A1.    As you can see there, it is `this`(=Cache)'s `request response list`.
My PR just did it without using `this` argument and `promise`.

Reply:
Thank you very much for the clear explanation! LGTM then.

Reply:
Thanks for the review.  Can I ask you to stamp approved?